packages/calling-component-bindings/src/callUtils.ts

```diff
@@ -44,3 +46,38 @@ export const disposeAllLocalPreviewViews = async (callClient: StatefulCallClient
     await callClient.disposeView(undefined, undefined, view);
   }
 };
+
+/**
+ * Update the users displayNames based on the type of user they are
+ *
+ * @private
+ */
+export const updateUserDisplayNames = (
+  participants:
+    | {
+        [keys: string]: RemoteParticipantState;
+      }
+    | undefined
+): RemoteParticipantState[] => {
+  let newParticipants: RemoteParticipantState[] = [];
+  if (participants) {
+    newParticipants = memoizedUpdateDisplayName((memoizedFn) => {
+      return Object.values(participants).map((p) => {
+        const pid = toFlatCommunicationIdentifier(p.identifier);
+        return memoizedFn(pid, p);
+      });
+    });
+  }
+  return newParticipants;
```

Comment:
This is going to return a new array each time (the contents of the array has been memoized but not the array itself)